docs/concepts.rst

```diff
@@ -271,3 +271,89 @@ Modules not yet Supported for Criteria
 :mod:`RunHistory <cbc_sdk.audit_remediation.base.RunHistory>`
 :mod:`Event <cbc_sdk.enterprise_edr.base.Event>`
 :mod:`Process <cbc_sdk.enterprise_edr.base.Process>`
+
+
+Asynchronous queries
+--------------------
+
+A number of queries allow for asynchronous mode of operation. Those utilize python futures and the request itself is performed in a separate worker thread.
+An internal thread pool is utilized to support multiple CBC queries executing in an asynchronous manner without blocking the main thread.
+
+Execute an asynchronous query
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Running asynchronous queries is done by invoking the ``execute_async()`` method, e.g:
+
+  >>> async_query = api.select(EnrichedEvent).where('process_name:chrome.exe')
+
+The ``execute_async()`` method returns a python future object that can be later on waited for results.
+
+Fetching asynchronous queries' results
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Results from asynchronous queries can be retrieved by using the result() method since they are actually futures:
+
+  >>> print(async_query.result())
+
+This would block the main thread until the query completes.
+
+Modules with support for asynchronous queries
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:mod:`Process <cbc_sdk.enterprise_edr.Process>`
+
+:mod:`EnrichedEvent <cbc_sdk.endpoint_standard.EnrichedEvent>`
+
+:mod:`EnrichedEventFacet <cbc_sdk.endpoint_standard.EnrichedEventFacet>`
+
+
+Facets
+------
+
+Facet search queries return statistical information indicating the relative weighting of the requested values as per the specified criteria.
+There are two types of criteria that can be set, one is the ``range`` type which is used to specify discrete values (integers or timestamps - specified both as seconds since epoch and also as ISO 8601 strings).
+The results are then grouped by occurence within the specified range.
+The other type is the ``term`` type which allow for one or more fields to use as a criteria on which to return weighted results.
+
+Setting ranges
+^^^^^^^^^^^^^^
+
+Ranges are configured via the ``add_range()`` method which accepts a dictionary of range settings:
```

Comment:
Out of curiosity why are we passing ranges as a dictionary instead of requiring the four properties in the function call? Using a dictionary makes it less intuitive when you look at the function call. Are we gaining a benefit somewhere?

Reply:
This comes from the base class, FacetQuery, the reason being we can add either a single range or a list of ranges in one call as opposed to separate calls to add individual ranges. We do the same for terms. It would be problematic to add a list of criteria as function arguments. I think this might need a discussion (ATM both PRs aren't merged yet).

Reply:
That's not how this comes across though it looks like it only accepts a dictionary not an array of dictionaries

Reply:
Got it. Corrected.